cowamm/kpi/competitors/curve/curve_kpis_4232873.sql

```diff
@@ -0,0 +1,25 @@
+-- Computes volume, tvl and APR for Curve pools
+-- APR is measured as the fees earned per $ invested, over the last 24 hours, projected over 1 year
+-- Parameters:
+-- {{blockchain}}: The blockchain to query
+-- {{end_time}}: The end time of the time window (end_time - 1 day; end_time), defaults to now()
+select
+    r1.contract_address,
+    r1.fee,
+    r1.tvl,
+    sum(amount_usd) as volume,
+    365 * sum(amount_usd * r.fee / r.tvl) as apr
+from "query_4232976(blockchain='{{blockchain}}')" as r
+inner join "query_4232976(blockchain='{{blockchain}}')" as r1
+    on
+        r.contract_address = r1.contract_address
+left join curve.trades as t
+    on
+        r.contract_address = t.project_contract_address
+        and r.tx_hash = t.tx_hash
+where
+    t.block_time >= date_add('day', -1, (case when '{{end_time}}' = '2024-01-01' then now() else timestamp '{{end_time}}' end))
+    and r1.latest = 1
```

Comment:
4232976 already filters for `latest = 1` doesn't it?

Reply:
No, it might feel confusing in the last part of the code: it filters by latest_per_pool only to find the largest pools, then it includes all of the tvls after every event which happened

`select * from recent_tvl
where contract_address in (
    select contract_address
    from recent_tvl
    where latest_per_pool = 1
    order by tvl desc
    limit {{number_of_pools}}
)
`